web/models/VagaBeneficioDTO.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+abstract class VagaBeneficioDTO implements DTOInterface
+{
+    use DbTrait;
+
+    public static function salvar($vagaBeneficio){
+        $pdo = static::conectarDB();
+
+        if (empty($vagaBeneficio->getVagaId()) && empty($vagaBeneficio->getBeneficioId())) {
```

Comment:
Essa checagem não vai funcionar por que os IDs são de outras tabelas, eles sempre vão existir mesmo quando for um insert.

Aqui você tem que dar um select na tabela vendo se já existe um registro pra esse beneficio e essa vaga. 
Se existir é UPDATE, se nao é INSERT.